include/triton/Tools/PluginUtils.h

```diff
@@ -14,32 +15,42 @@ enum TritonPluginResult {
 };
 #define TRITON_PLUGIN_API                                                      \
   extern "C" __attribute__((visibility("default"))) TritonPluginResult
+#define TRITON_PLUGIN_API_TYPE(_TYPE)                                          \
+  extern "C" __attribute__((visibility("default"))) _TYPE
 
 struct TritonPlugin {
   TritonPlugin() = delete;
   TritonPlugin(std::string filename) : filename(filename) {}
 
+public:
+  llvm::Error checkLibraryValid(const std::string &error) const;
+  static constexpr char *ENUMERATE_PASSES =
+      (char *)"tritonEnumeratePluginPasses";
```

Comment:
Using a `char[]` can be nicer because the type then carries information about the length ([example](https://godbolt.org/z/KKPnv1q99)). It would also let you get rid of the risky `const char * -> char *` cast.

```suggestion
  static constexpr char ENUMERATE_PASSES[] = "tritonEnumeratePluginPasses";
```

If you want to keep it as a plain pointer, you can still get rid of the cast by making it a `static constexpr const char *`.